src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/Blocks/tabs.html.twig

```diff
@@ -34,7 +34,7 @@
 {% set isPriceTabValid = productForm.price.vars.valid %}
 {% set isShippingTabValid = productForm.shipping.vars.valid %}
 {% set isSEOTabValid = productForm.seo.vars.valid and productForm.redirect_option.vars.valid %}
-{% set isStockTabValid = productForm.stock.vars.valid %}
+{% set isStockTabValid = productForm.stock is not defined or productForm.stock.vars.valid %}
```

Comment:
So it means we don't handle any quantities for virtual products?

Reply:
we will have stocks inside the virtual product tab, but its not the "stock" tab anymore - it is virtual product tab.
Same with combinations, we will have "combinations" tab, but we also will need a little part of stocks to be reused there, but anyway it will still be the "combinations" tab and not the 'stocks" tab anymore

Reply:
Mmm, I see so your approach would be to duplicate the stock part into several sub forms?
- regular product
- virtual product
- combination

Although it makes sense for combinations since they all handle their individual stock, I still wonder if it's appropriate for virtual product. It could be displayed the same way as a regular product stock tab But just with the additional virtual file component

For now, from what I understand, the stock FormType is removed but the virtual tab only contains the Virtual file form, so we are losing all the stock feature for virtual file Unless I missed something

Reply:
Ok my bad now I see it's in the `virtual.html.twig` template!

Reply:
Ok I understand better now, my bad! So it works but it feels a bit messy to look for the form in different files 🤔  Especially now that we start having a few conditions depending on the product type

That's why I started to split all tabs in separate files so that they are smaller and easier to grasp, but it's still in this pending PR https://github.com/PrestaShop/PrestaShop/pull/23330/files#diff-b13a2b63998e59e446205675701dea5009033693c58e7f512dd3c3571fa2579b

But we don't need to wait for it, it's just my opinion but I fell like it would be cleaner if we had one template per tab, and keep stock and virtual file in the same unique tab template (with a conditional display for virtual file). This would make the `edit.html.twig` simpler because the condition would wimply be:

```twig
      {% if productType == 'combinations' %}
        {% block product_tab_combinations %}
          {{ include('@PrestaShop/Admin/Sell/Catalog/Product/Tabs/combinations.html.twig') }}
        {% endblock %}
      {% else %}
        {% block product_tab_stock %}
          {{ include('@PrestaShop/Admin/Sell/Catalog/Product/Tabs/stock.html.twig') }}
        {% endblock %}
      {% endif %}
```

Although `stock.html.twig` name can probably be improved, it seems more stable to have only one template that displays `{{ form_widget(productForm.stock) }}` than two different templates, with two or three conditions in each